framework/include/materials/MaterialBase.h

```diff
@@ -353,6 +391,26 @@ MaterialBase::getGenericZeroMaterialProperty(const std::string & prop_name)
   return preload_with_zero;
 }
 
+template <typename T, bool is_ad>
+const GenericMaterialProperty<T, is_ad> &
+MaterialBase::getGenericZeroMaterialProperty()
+{
+  // static zero property storage
+  static GenericMaterialProperty<T, is_ad> zero;
+
+  // resize to accomodate maximum number of qpoints
+  // (in multiapp scenarios getMaxQps can return different values in each app; we need the max)
+  unsigned int nqp = _fe_problem.getMaxQps();
+  if (nqp > zero.size())
+    zero.resize(nqp);
```

Comment:
Do you need to do all this? I guess maybe so if you plan on using the zero material property, but generally I use zero material property for switch cases in the header. I suppose this is safest.

Reply:
I definitely need this, as the zero property is used in the derivative material interface to return zero derivatives in case they do not exist. This would leas to inexact Jacobians or the assumption of constant variables 9with vanishing derivatives).

Reply:
ya good point